nat-lab/tests/utils/connection_tracker.py

```diff
@@ -174,30 +235,41 @@ async def get_out_of_limits(self) -> Optional[Dict[str, int]]:
 
         return out_of_limit_connections if bool(out_of_limit_connections) else None
 
-    async def synchronize(self):
+    def _check_tcp_state_for_events(self, conntracker_event: ConntrackerEvent) -> None:
+        if tcp_state := conntracker_event.tcp_state:
+            if notify_events := self._tcp_state_events.get(tcp_state):
+                for event in notify_events:
+                    # set only the first unset event
+                    if not event.is_set():
+                        print(datetime.now(), "ConnectionTracker setting event", event)
+                        event.set()
+                        break
```

Comment:
Coninuation of the defaultdict nit. Another nice benefit is that you can simplify this to:
```
if tcp_state := conntracker_event.tcp_state:
    try:
        self._tcp_state_events[tcp_state].pop(0).set()
    except IndexError:
        pass
```